cfn-resources/search-index/CHANGELOG.md

```diff
@@ -0,0 +1,49 @@
+# Changelog
+
+## (2023-10-12)
+
+**(POSSIBLE CHANGE) FIXED MAPPING INPUTS**
+
+Originally, the mapping fields were inputted like this:
+``` json
+mappings: {
+fields: [
+"ownershipDetails.assetName:string"
+],
+},
+``` 
+
+the problem with this approach  is that the user will only be able to input one field and the time, and also only the type
+was specified, some data types like "autocomplete" may require extra parameters. (like analyzer, maxGrams, minGrams, etc),
+also a field would have accepted only one data type, preventing the user to setting multiple types to a single field. 
+
+we required a way to input dynamic structures to each property, but since cloud formation does not provide a way to
+input dynamic objects as parameters, we decided to input the data type list as a stringify json,
+representing a list of types like is shown in the MongoDB atlas UI
+
+Original:
+``` json
+mappings: {
+fields: [
+"property_type:string"
+],
+},
+``` 
+
+New:
+``` json
+mappings: {
+fields: [
+"property_type:[{\"type\":\"autocomplete\"},{\"type\":\"string\"}}]"
```

Comment:
I find it confusing that the string value has a mix between non json `property_type:` and then a proper json format `[{\"type\":\"autocomplete\"},{\"type\":\"string\"}}]`.
Looking into how we define `mappings_fields` in terraform, it has a single property that expects a json format in its entirety to define all mappings: https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/search_index#mappings_fields
I think this will be more clear to the end user, what do you think?

Reply:
so this is my original approach, the mappings is a json file, the problem is that if i implement it like this, im not able to update an existing stack. for example:
if an user is using search index with a list of strings (asi it is right now), when he tries to use the new version, and execute the update, this will throw an error, because cfn will try to populate the previous model with an interface{} and will fail

Reply:
> so this is my original approach, the mappings is a json file, the problem is that if i implement it like this, im not able to update an existing stack

Have you considered a new field which allows defining a single string with all the json configuration? And we can deprecate the old `fields` property but leave it available to avoid any breaking changes. These 2 fields could be mutually exclusive, meaning users would use one approach or the other. What do you think? I would imagine user could update the stack from using one field to another as long as the underlying content is the same.